map_test.go

```diff
@@ -1974,6 +1974,78 @@ func BenchmarkMap(b *testing.B) {
 	})
 }
 
+func BenchmarkIterate(b *testing.B) {
+	m, err := NewMap(&MapSpec{
+		Type:       Hash,
+		KeySize:    8,
+		ValueSize:  8,
+		MaxEntries: 1000,
+	})
+	if err != nil {
+		b.Fatal(err)
+	}
+	b.Cleanup(func() {
+		m.Close()
+	})
+
+	var (
+		n      = m.MaxEntries()
+		keys   = make([]uint64, n)
+		values = make([]uint64, n)
+	)
+
+	for i := 0; uint32(i) < n; i++ {
+		keys[i] = uint64(i)
+		values[i] = uint64(i)
+	}
+
+	if _, err := m.BatchUpdate(keys, values, nil); err != nil {
+		b.Fatal(err)
+	}
+
+	b.Run("MapIterator", func(b *testing.B) {
+		var k, v uint64
+
+		b.ReportAllocs()
+
+		for i := 0; i < b.N; i++ {
+			var count uint64
+			iter := m.Iterate()
+			for iter.Next(&k, &v) {
+				count++
+			}
+			if err := iter.Err(); err != nil {
+				b.Fatal(err)
+			}
+			if count != uint64(m.MaxEntries()) {
```

Comment:
How about skipping this? Benchmarky code doesn't really benefit from such error checking imo.

Reply:
Yeah, fair point. It was only useful while writing it.